spring-integration-core/src/main/java/org/springframework/integration/handler/AbstractMessageProducingHandler.java

```diff
@@ -336,4 +316,41 @@ protected boolean shouldCopyRequestHeaders() {
 		return true;
 	}
 
+	protected Object resolveErrorChannel(final MessageHeaders requestHeaders) {
+		Object errorChannel = requestHeaders.getErrorChannel();
+		if (errorChannel == null) {
+			try {
+				errorChannel = getChannelResolver().resolveDestination("errorChannel");
+			}
+			catch (DestinationResolutionException e) {
+				// ignore
+			}
+		}
+		return errorChannel;
+	}
+
+	protected void sendErrorMessage(final Message<?> requestMessage, Throwable ex) {
+		Object errorChannel = resolveErrorChannel(requestMessage.getHeaders());
```

Comment:
The proper convention to declare used method after the user.
Minor, of course, but for "heads up!"  anyway.